### DISCUSSION TITLE
Help me compile it on Cygwin

### DISCUSSION THREAD

**Discussion by thjwong** [sm00th/bitlbee-discord#225] [Q&A] [answered]
I made it work on Devuan 64-bit (not without the token hassle) and now trying to compile it also for Windows 10 with Cygwin x64.

First off, bitlbee on Github does not install the libraries properly with `make install-dev`.
And after copying all the libraries or fixing the path to libraries problem, I have this:

> `my_computer_username@localhostname:~/git/bitlbee-discord$ make`
`make  all-recursive`
`make[1]: Entering directory '/cygdrive/d/home/my_computer_username/git/bitlbee-discord'`
`Making all in src`
`make[2]: Entering directory '/cygdrive/d/home/my_computer_username/git/bitlbee-discord/src'`
 ` CC       discord_la-discord.lo`
`In file included from ./bitlbee.h:135:0,`
 `                from discord.h:20,`
 `                from discord.c:18:`
`/usr/include/w32api/storage.h:7:2: warning: #warning Your code should include objbase.h instead of storage.h header file. [-Wcpp]`
`
 #warning Your code should include objbase.h instead of storage.h header file.
  ^~~~~~~
`
`In file included from discord.h:20:0,`
 `                from discord.c:18:`
`/usr/lib/gcc/x86_64-pc-cygwin/7.4.0/include/mm_malloc.h: In function ‘_mm_malloc’:`
`./bitlbee.h:68:25: warning: implicit declaration of function `
 `‘__PLEASE_USE_THE_GLIB_MEMORY_ALLOCATION_SYSTEM__’ [-Wimplicit-function-declaration]
 #define malloc          __PLEASE_USE_THE_GLIB_MEMORY_ALLOCATION_SYSTEM__`
`
                         ^`
`In file included from ./auth.h:4:0,`
`                 from ./bitlbee.h:136,`
`                 from discord.h:20,`
`                 from discord.c:18:
`
`/usr/include/w32api/storage.h: At top level:`
`/usr/include/w32api/storage.h:7:2: warning: #warning Your code should include objbase.h instead of storage.h header file. [-Wcpp]`
` #warning Your code should include objbase.h instead of storage.h header file.`
`  ^~~~~~~`
`In file included from ./bitlbee.h:136:0,`
`                 from discord.h:20,`
`                 from discord.c:18:`
`./auth.h:8:2: error: expected specifier-qualifier-list before ‘storage_status_t’`
`  storage_status_t (*check_pass)(const char *nick, const char *password);`
`  ^~~~~~~~~~~~~~~~`
`./auth.h:12:1: error: unknown type name ‘storage_status_t’; did you mean ‘error_status_t’?`
` storage_status_t auth_check_pass(irc_t *irc, const char *nick, const char *password);`
` ^~~~~~~~~~~~~~~~`
` error_status_t`
`make[2]: *** [Makefile:490: discord_la-discord.lo] Error 1`
`make[2]: Leaving directory '/cygdrive/d/home/my_computer_username/git/bitlbee-discord/src'`
`make[1]: *** [Makefile:427: all-recursive] Error 1`
`make[1]: Leaving directory '/cygdrive/d/home/my_computer_username/git/bitlbee-discord'`
`make: *** [Makefile:359: all] Error 2`

**Comment by sm00th** [comment, marked as answer]
All of the issues seem to be leading to bitlbee's headers so whatever tricks you used to compile bitlbee on cygwin should also apply here.
